spec/chartmogul/subscription_spec.rb

```diff
@@ -176,5 +178,13 @@
         expect(subject.cancellation_dates).to eq [Time.utc(1999, 12, 31, 22)]
       end
     end
+
+    describe 'when array has time objects' do
+      let(:dates) { [Time.utc(2000, 1, 1)] }
+
+      it 'is setting the cancellation dates of the subscription' do
+        expect { subject }.to raise_error(TypeError, 'no implicit conversion of Time into String')
```

Comment:
I think the expectation should be that it works. In ruby, developers work with dates as objects, not text strings, after all that's also how we suggest it in docs:
`subscription.set_cancellation_dates([Time.utc(2016, 1, 1), Time.utc(2017, 1, 1)])`

Suggestion: either make it accept only `Time` objects or make the `parse_dates` iterate with `if string -> parse`

Reply:
Ok, it makes sense.